sirepo/package_data/template/activait/train.py.jinja

```diff
@@ -35,7 +36,36 @@ model.fit(
 testy = test[:, out_idx]
 testx = test[:, in_idx]
 
+{% endif %}
+
+{% if image_data %}
+def _domain_len(outputs_true):
+    return len(numpy.unique(outputs_true))
+
+def _prediction_shape(prediction, testy):
+    return prediction.shape[-1] == _domain_len(testy)
+
+def _classification_case(prediction, testy):
+    return {{ discreteOutputs }} and _prediction_shape(prediction, testy)
+
+def _predictions_final(p, testy):
+    if _classification_case(p, testy):
+        p = np.argmax(p, axis=1)
+    return p
```

Comment:
I think it's best to avoid changing the input parameter, so:
```python
def _predictions_final(p, testy):
    return np.argmax(p, axis=1) if _classification_case(p, testy) else p
```

Reply:
Good call